sparse_list.py

```diff
@@ -144,13 +146,13 @@ def __initialise_from_iterable(self, arg):
         self.size = len(self.elements)
 
     def __eq__(self, other):
-        return all(a == b for a, b in zip_longest(self, other))
+        return len(self) == len(other) and all(a == b for a, b in zip_longest(self, other))
 
     def __ne__(self, other):
         return not self.__eq__(other)
 
     def __lt__(self, other):
-        return any(a < b for a, b in zip_longest(self, other))
+        return list(self) < list(other)
```

Comment:
It's unclear which unit test this fixes.

Could we not short-circuit the evaluation on size of self _vs_ other and keep the rest of the evaluation lazy?

Reply:
The problem is that `any(a < b for a, b in zip_longest(self, other))` is not the lexical order that `list` uses… Let me add a test for that.

Reply:
Thanks for the test, that made it a lot clearer.

Converting two whole SparseLists to lists for this comparison seems expensive. Should we do this instead?

```suggestion
        for a, b in zip_longest(self, other):
            if a < b:
                return True
            if a > b:
                return False
        return False
```